### PR TITLE
Support subscription selection for interactive login process

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Accounts/Authentication.ResourceManager/Models/PSAzureProfile.cs [hunk 1]
```diff
@@ -84,10 +84,8 @@ public class PSAzureProfile
         /// <summary>
         /// The current credentials and metadata for connecting with the current Azure cloud instance.
         /// </summary>
-        [Ps1Xml(Label = "Account", Target = ViewControl.Table, ScriptBlock = "$_.Context.Account.ToString()", Position = 0)]
-        [Ps1Xml(Label = "SubscriptionName", Target = ViewControl.Table, ScriptBlock = "$_.Context.Subscription.Name", Position = 1)]
-        [Ps1Xml(Label = "TenantId", Target = ViewControl.Table, ScriptBlock = "$_.Context.Tenant.ToString()", Position = 2)]
-        [Ps1Xml(Label = "Environment", Target = ViewControl.Table, ScriptBlock = "$_.Context.Environment.ToString()", Position = 3)]
+        [Ps1Xml(Label = "Subscription name", Target = ViewControl.Table, ScriptBlock = "$_.Context.Subscription.Name", Position = 0)]
+        [Ps1Xml(Label = "Tenant domain name", Target = ViewControl.Table, ScriptBlock = "$_.Context.Tenant.DefaultDomain", Position = 1)]
```
Comment (msJinLei) [inline src/Accounts/Authentication.ResourceManager/Models/PSAzureProfile.cs:88]: Subscription Name and Default Domain can be null

File: src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs [hunk 2]
```diff
@@ -608,9 +620,9 @@ private string PreProcessAuthScope()
             return mappedScope;
         }
 
-        private bool IsUsingInteractiveAuthentication()
+        private bool IsBrowserPopUpInteractiveAuthentication()
         {
-            return ParameterSetName == UserParameterSet && UseDeviceAuthentication == false;
+            return ParameterSetName.Equals(UserParameterSet) && UseDeviceAuthentication.IsPresent == false;
```
Comment (msJinLei) [inline src/Accounts/Accounts/Account/ConnectAzureRmAccount.cs:625]: What is the difference from "ParameterSetName == UserParameterSet && UseDeviceAuthentication == false"

File: src/Accounts/Accounts/Models/RMProfileClient.cs [hunk 3]
```diff
@@ -296,11 +316,59 @@ private IAzureTenant GetDetailedTenantFromQueryHistory(List<AzureTenant> tenants
...
+                    if (shouldWriteSubscriptionSelectionTable && markDefaultSubscription)
+                    {
+                        WriteInformationMessage($"{Environment.NewLine}The default is marked with an {Foreground.White}*{PSStyle.Reset}; " +
+                            $"the default tenant is '{GetDetailedTenantFromQueryHistory(queriedTenants, lastUsedSubscription?.GetProperty(AzureSubscription.Property.Tenants))?.GetProperty(AzureTenant.Property.DefaultDomain)}' " +
+                            $"and subscription is '{lastUsedSubscription?.Name} ({lastUsedSubscription?.Id})'.");
+                    }
+
+                    if (shouldWriteSubscriptionSelectionTable && null != prompt)
+                    {
+                        string input = markDefaultSubscription ? prompt($"{Environment.NewLine}Select a tenant and subscription (type a number or Enter to accept default): ") : prompt($"{Environment.NewLine}Select a tenant and subscription: ");
```
Comment (msJinLei) [inline src/Accounts/Accounts/Models/RMProfileClient.cs:342]: Use resource string

Comment (msJinLei) [inline src/Accounts/Accounts/Models/RMProfileClient.cs:342]: Use resource strings

**Review by msJinLei — DISMISSED** [review]
Won't review for now and just approve it

**Comment by vidai-msft** [comment]
@BethanyZhou Please fix the test error on CI

File: src/Accounts/Accounts/Models/RMProfileClient.cs (hunk 3 above)
Comment (BethanyZhou) [reply src/Accounts/Accounts/Models/RMProfileClient.cs:342]: Will do for public release.

**Pull request closed without merging**